### PR TITLE
[Type checker] Look through optionals when checking for rethrows.

### PR DESCRIPTION
**Pull request by DougGregor**
Fixes SR-5427.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Review by huonw — DISMISSED** [review]

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
My earlier fix was incorrect. Asking for a re-review

File: lib/Sema/TypeCheckError.cpp [hunk 1]
```diff
@@ -134,6 +134,9 @@ class AbstractFunction {
         fn = conversion->getSubExpr()->getValueProvidingExpr();
       } else if (auto conversion = dyn_cast<BindOptionalExpr>(fn)) {
         fn = conversion->getSubExpr()->getValueProvidingExpr();
+      // Look through optional injections
```
Comment (rjmccall) [inline lib/Sema/TypeCheckError.cpp]: For consistency, this should end in a period.

**Review by rjmccall — APPROVED** [review]

File: lib/Sema/TypeCheckError.cpp (hunk 1 above)
Comment (DougGregor) [reply lib/Sema/TypeCheckError.cpp]: Indeed. I'll tack that on in some other random PR I land today

**Pull request merged**